python_modules/libraries/dagster-tableau/dagster_tableau/translator.py

```diff
@@ -147,13 +147,21 @@ def get_asset_spec(self, data: TableauTranslatorData) -> AssetSpec:
     def get_sheet_asset_key(self, data: TableauTranslatorData) -> AssetKey:
         return self.get_sheet_spec(data).key
 
+    """ If published data sources are available (i.e., parentPublishedDatasources exists and is not empty), it means you can form the lineage by using the luid of those published sources.
+    If the published data sources are missing, you create assets for embedded data sources by using their id.
+    """
     def get_sheet_spec(self, data: TableauTranslatorData) -> AssetSpec:
         sheet_embedded_data_sources = data.properties.get("parentEmbeddedDatasources", [])
-        data_source_ids = {
-            published_data_source["luid"]
-            for embedded_data_source in sheet_embedded_data_sources
-            for published_data_source in embedded_data_source.get("parentPublishedDatasources", [])
-        }
+
+        data_source_ids = set()
+        for embedded_data_source in sheet_embedded_data_sources:
+            published_data_source_list = embedded_data_source.get("parentPublishedDatasources", [])
+            is_publish_data_source_absent = True
+            for published_data_source in published_data_source_list:
+                is_publish_data_source_absent = False
+                data_source_ids.add(published_data_source["luid"])
+            if is_publish_data_source_absent:
```

Comment:
You can do this instead of using an additional flag:

```suggestion
            if not published_data_source_list:
```